test_cases/README.md

```diff
@@ -1,31 +1,35 @@
 ## Regression tests for typeshed
 
-This directory contains regression tests for the stubs found elsewhere in the
-typeshed repo.
+This directory contains code samples that act as a regression test for the
+standard-library stubs found elsewhere in the typeshed repo.
 
-**This directory should *only* contain tests for functions and classes which
+**This directory should *only* contain test cases for functions and classes which
 are known to have caused problems in the past, where the stubs are difficult to
 get right.** 100% test coverage for typeshed is neither necessary nor
 desirable, as it would lead to code duplication. Moreover, typeshed has
 multiple other mechanisms for spotting errors in the stubs.
 
 ### The purpose of these tests
 
-Different tests in this directory serve different purposes. For some stubs in
+Different test cases in this directory serve different purposes. For some stubs in
 typeshed, the type annotations are complex enough that it's useful to have
 basic sanity checks that test whether a type checker understands the intent of
 the annotations correctly. Examples of tests like these are
 `stdlib/builtins/test_pow.py` and `stdlib/asyncio/test_gather.py`.
 
-Other tests, such as the tests for `ExitStack` in `stdlib/test_contextlib.py`
-and the tests for `LogRecord` in `stdlib/test_logging.py`, do not relate to
+Other test cases, such as the samples for `ExitStack` in `stdlib/test_contextlib.py`
+and the samples for `LogRecord` in `stdlib/test_logging.py`, do not relate to
 stubs where the annotations are particularly complex, but they *do* relate to
 stubs where decisions have been taken that might be slightly unusual. These
-tests serve a different purpose: to check that type checkers do not emit
+test cases serve a different purpose: to check that type checkers do not emit
 false-positive errors for idiomatic usage of these classes.
 
 ### How the tests work
 
+None of the code in this directory is actually executed in typeshed's CI.
```

Comment:
I think this sentence is a bit confusing. Maybe something along the lines of "The code in this directory is not meant to be executed."